onetl/connection/file_connection/samba.py

```diff
@@ -0,0 +1,292 @@
+                if entry.filename not in {".", ".."}  # Filter out '.' and '..'
+            ]
+        return [self.client.getAttributes(self.share, os.fspath(path))]
+
+    def _extract_name_from_entry(self, entry) -> str:
+        return entry.filename
+
+    def _is_dir_entry(self, top: RemotePath, entry) -> bool:
+        return entry.isDirectory
+
+    def _is_file_entry(self, top: RemotePath, entry) -> bool:
+        return not entry.isDirectory
+
+    def _extract_stat_from_entry(self, top: RemotePath, entry) -> RemotePathStat:
+        if entry.isDirectory:
+            return RemotePathStat(st_mode=stat.S_IFDIR)
+
+        return RemotePathStat(
+            st_size=entry.file_size,
+            st_mtime=entry.last_write_time,
+            st_uid=entry.filename,
+        )
+
+    def _get_client(self) -> SMBConnection:
+        is_direct_tcp = self.protocol == "SMB"
+        use_ntlm_v2 = self.auth_type == "NTLMv2"
+        conn = SMBConnection(
+            username=self.user,
+            password=self.password.get_secret_value() if self.password else None,
+            my_name="optional_client_name",
+            remote_name=self.host,
+            domain=self.domain,
+            use_ntlm_v2=use_ntlm_v2,
+            sign_options=2,
+            is_direct_tcp=is_direct_tcp,
+        )
+        conn.connect(self.host, port=self.port)
+        return conn
+
+    def _is_client_closed(self, client: SMBConnection) -> bool:
+        try:
+            socket_fileno = client.sock.fileno()
+        except (AttributeError, OSError):
+            return True
+
+        return socket_fileno == -1
+
+    def _close_client(self, client: SMBConnection) -> None:
+        self.client.close()
+
+    def _download_file(self, remote_file_path: RemotePath, local_file_path: LocalPath) -> None:
+        with open(local_file_path, "wb") as local_file:
+            self.client.retrieveFile(
+                self.share,
+                os.fspath(remote_file_path),
+                local_file,
+            )
+
+    def _get_stat(self, path: RemotePath) -> RemotePathStat:
+        info = self.client.getAttributes(self.share, os.fspath(path))
+
+        if self.is_dir(os.fspath(path)):
+            return RemotePathStat(st_mode=stat.S_IFDIR)
+
+        return RemotePathStat(
+            st_size=info.file_size,
+            st_mtime=info.last_write_time,
+            st_uid=info.filename,
+        )
+
+    def _remove_file(self, remote_file_path: RemotePath) -> None:
+        self.client.deleteFiles(
+            self.share,
+            os.fspath(remote_file_path),
+        )
+
+    def _create_dir(self, path: RemotePath) -> None:
+        path_obj = Path(path)
+        for parent in reversed(path_obj.parents):
+            # create dirs sequentially as .createDirectory(...) cannot create nested dirs
+            try:
+                self.client.createDirectory(self.share, os.fspath(parent))
+            except OperationFailure:
+                pass
```

Comment:
```suggestion
            try:
                self.client.getAttributes(self.share, os.fspath(path))
            except OperationFailure:
                self.client.createDirectory(self.share, os.fspath(parent))
```
Иначе если есть проблема при создании какой-то из родительских папок, мы никогда не узнаем, в чем она была, только увидим сообщение об ошибке, что родительская папка не существует.